### PR TITLE
Implementation of LiftAll for Coproduct

### PR DESCRIPTION
**Pull request by AlecZorab**
Straightforward translation of `shapeless.ops.hlist.LiftAll`

### CODE REVIEW

**Comment by AlecZorab** [comment]
Now with extra added tests

**Comment by AlecZorab** [comment]
I'm not sure exactly why codecov is complaining about this. If someone can point me at what I've done wrong, I'll fix it.

**Comment by milessabin** [comment]
I think this is actually an issue with the scoverage SBT plugin. For the time being I'm ignoring the complaints.

**Comment by milessabin** [comment]
LGTM ... many thanks ... merging :-)

**Comment by Atry** [comment]
Is it possible to make `LiftAll` a `DepFn0`?

File: core/src/main/scala/shapeless/ops/coproduct.scala [hunk 1]
```diff
@@ -1268,4 +1268,30 @@ object coproduct {
         def apply(): Out = wh.value :: rt()
       }
   }
+
+  sealed trait LiftAll[F[_], In <: Coproduct] {
```
Comment (Atry) [inline core/src/main/scala/shapeless/ops/coproduct.scala]: Why not extends `DepFn0`?

Comment (milessabin) [inline core/src/main/scala/shapeless/ops/coproduct.scala]: Yes, I think that would be preferable. PR?

Comment (AlecZorab) [inline core/src/main/scala/shapeless/ops/coproduct.scala]: The main reason for not extending `DepFn0` was that I didn't know it existed. Probably want the sam to be treatment on `HList`

**Pull request merged**